### PR TITLE
Optimize the error message of OP.

### PR DESCRIPTION
**Pull request by xiemoyuan**
### PR types

Others

### PR changes

OPs

### Describe

Optimize the error message in `paddle/fluid/operators/gather_tree_op.cc`

### CODE REVIEW

File: paddle/fluid/operators/gather_tree_op.cc [hunk 1]
```diff
@@ -22,18 +22,19 @@ class GatherTreeOp : public framework::OperatorWithKernel {
...
+                      "Output(Out) of GatherTreeOp should not be null.");
 
     auto ids_dims = ctx->GetInputDim("Ids");
     auto parents_dims = ctx->GetInputDim("Parents");
-    PADDLE_ENFORCE(ids_dims == parents_dims,
-                   "The shape of Input(Parents) must be same with the shape of "
-                   "Input(Ids).");
+    PADDLE_ENFORCE_EQ(
+        ids_dims == parents_dims, true,
+        "The shape of Input(Parents) must be same with the shape of "
```
Comment (guoshengCS) [inline paddle/fluid/operators/gather_tree_op.cc]: 请注意这里使用`platform::errors::InvalidArgument`，参考https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/batch_norm_op.cc#L54

File: paddle/fluid/operators/gather_tree_op.cc [hunk 2]
```diff
@@ -22,18 +22,19 @@ class GatherTreeOp : public framework::OperatorWithKernel {
   using framework::OperatorWithKernel::OperatorWithKernel;
 
   void InferShape(framework::InferShapeContext* ctx) const override {
-    PADDLE_ENFORCE(ctx->HasInput("Ids"),
-                   "Input(Ids) of GatherTreeOp should not be null.");
-    PADDLE_ENFORCE(ctx->HasInput("Parents"),
-                   "Input(Parents) of GatherTreeOp should not be null.");
-    PADDLE_ENFORCE(ctx->HasOutput("Out"),
-                   "Output(Out) of GatherTreeOp should not be null.");
+    PADDLE_ENFORCE_EQ(ctx->HasInput("Ids"), true,
```
Comment (guoshengCS) [inline paddle/fluid/operators/gather_tree_op.cc]: 这里可以使用`OP_INOUT_CHECK `，可以参考这里 https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/batch_norm_op.cc#L28

File: paddle/fluid/operators/gather_tree_op.cc (hunk 1 above)
Comment (xiemoyuan) [reply paddle/fluid/operators/gather_tree_op.cc]: Done

Comment (xiemoyuan) [reply paddle/fluid/operators/gather_tree_op.cc]: Done. Thanks.

File: paddle/fluid/operators/gather_tree_op.cc (hunk 2 above)
Comment (xiemoyuan) [reply paddle/fluid/operators/gather_tree_op.cc]: Done. Thanks.

**Review by guoshengCS — APPROVED** [review]

**Pull request merged**